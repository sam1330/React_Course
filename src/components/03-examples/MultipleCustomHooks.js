import React from "react";

import { useFetch } from "../../hooks/useFetch";

import "./effects.css"
const MultipleCustomHooks = () => {

  const state = useFetch(`https://www.breakingbadapi.com/api/quotes/1`);

  console.log(state);
  return (
    <>
      <h1>Custom Hooks</h1>
    </>
  )
};

export default MultipleCustomHooks;