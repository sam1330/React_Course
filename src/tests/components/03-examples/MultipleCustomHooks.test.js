import React from "react";
import { shallow } from "enzyme";

import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";

jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("tests on <MultipleCustomHooks />", () => {

  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => { }
    });
  });

  test('should show component correctly', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();

  });


  test('should show info correctly', () => {
    useFetch.mockReturnValue({
      data: [{
        author: "Sammy",
        quote: "Hello world",
      }],
      loading: false,
      error: null
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find('.alert').exists()).toBe(false);
    expect(wrapper.find('.mb-0').text().trim()).toBe('Hello world');
    expect(wrapper.find('footer').text().trim()).toBe('Sammy');

  });
});