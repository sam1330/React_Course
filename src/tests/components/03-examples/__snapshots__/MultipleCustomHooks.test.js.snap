// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <MultipleCustomHooks /> should show component correctly 1`] = `
<Fragment>
  <h1>
    BrakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</Fragment>
`;
